eZ/Publish/API/Repository/Tests/ObjectStateServiceTest.php

```diff
@@ -1017,6 +1022,188 @@ public function testLoadObjectStateThrowsNotFoundException()
     }
 
     /**
+     * Data provider for PrioritizedLanguageList tests.
+     *
+     * @return array
+     */
+    public function getPrioritizedLanguagesList()
+    {
+        return [
+            [[], null],
+            [['eng-GB'], null],
+            [['eng-US'], 'eng-US'],
+            [['ger-DE'], 'ger-DE'],
+            [['eng-US', 'ger-DE'], 'eng-US'],
+            [['ger-DE', 'eng-US'], 'ger-DE'],
+            [['eng-GB', 'ger-DE', 'eng-US'], 'ger-DE'],
+        ];
+    }
+
+    /**
+     * Test that multi-language logic for loadObjectStateGroups respects prioritized language list.
+     *
+     * @dataProvider getPrioritizedLanguagesList
+     * @param string[] $prioritizedLanguages
+     * @param string|null $expectedLanguageCode
+     */
+    public function testLoadObjectStateGroupsWithPrioritizedLanguagesList(
+        array $prioritizedLanguages,
+        $expectedLanguageCode
+    ) {
+        // cleanup before the actual test
+        $this->deleteExistingObjectStateGroups();
```

Comment:
Note: already existing in initial data object state group messes with this test, so I removed it (just for this single test).
As an alternative, I could've improved initial data, but I found more valuable to test if what API just created aligns with what gets loaded.